tests/LinAlg/vectorTests.hpp

```diff
@@ -1085,6 +1085,98 @@ class VectorTests : public TestBase
     return reduceReturn(fail, &x);
   }
 
+  /**
+   * @brief Test: For addLinearDampingTerm method, which performs a "signed" axpy:
+   * `x[i] = alpha*x[i] + sign*ct` where sign=1 when exactly one of left[i] and right[i]
+   * is 1. and sign=0 otherwise.
+   */
+  bool vectorAddLinearDampingTerm(
+      hiop::hiopVector& x,
+      hiop::hiopVector& left,
+      hiop::hiopVector& right,
+      const int rank)
+  {
+    const local_ordinal_type N = getLocalSize(&x);
+    assert(N == getLocalSize(&left));
+    assert(N == getLocalSize(&right));
```

Comment:
I think it would be reasonable to just require size >=4 for this test rather than checking in the test and possibly leaving some corner cases untested.

```suggestion
  {
    const local_ordinal_type N = getLocalSize(&x);
    assert(N == getLocalSize(&left));
    assert(N == getLocalSize(&right));
    assert(N >= 4 && "Test `vectorAddLinearDampingTerm` must have size >=4 to test all use cases for this kernel");
```

Reply:
that way one cannot test vectors with unbalanced MPI distribution (as in potentially empty local data in an otherwise quite long vector)

Reply:
Okay, how should I run the test in order to test each case? I would naively just run the test on a vector with size >=4, so I think documenting how the function must be called in order to fully test would be helpful